### PR TITLE
Multi-Tenant Emails: Fixes for Studio login

### PR DESCRIPTION
**Pull request by OmarIthawi**
### Bug Description
When `APPSEMBLER_MULTI_TENANT_EMAILS` is enabled the `get_current_organization()` helper breaks. This PR fixes it by avoid calling it in Studio.

### Notable changes

When we do this another problem arises, Appsembler Academy trials won't be able to login in Studio. Therefore `MultipleObjectsReturned` will be raised. The solution is to reduce `MultipleObjectsReturned` errors in studio login by restricting access to course creators in Studio.

### Related Links

 - This PR is part of the larger [Multi-Tenant Emails proposal](https://github.com/appsembler/tech-design-proposals/blob/master/proposals/0003-tahoe-multi-tenant.md).
 - Jira: https://appsembler.atlassian.net/browse/RED-450

### CODE REVIEW

File: common/djangoapps/student/views/login.py [hunk 1]
```diff
@@ -153,6 +153,26 @@ def _log_failed_get_user_by_email(email):
         AUDIT_LOG.warning(u"Login failed - Unknown user email: {0}".format(email))
 
 
+def _is_in_lms():
+    """
+    Check if the function is executed in LMS.
```
Comment (johnbaldwin) [inline common/djangoapps/student/views/login.py:158]: There's nothing technically wrong with this statement, but when I first read it, It read like a test case to me.

Perhaps "Returns `True` if our caller is the LMS, `False` otherwise" ?

Maybe rephrase to "Check if the LMS is our caller"

File: common/djangoapps/student/views/login.py [hunk 2]
```diff
@@ -153,6 +153,26 @@ def _log_failed_get_user_by_email(email):
...
+    """
+    return settings.ROOT_URLCONF == 'lms.urls'
+
+
+def _get_course_creator_by_email(email):
+    """
+    Get a user by email with CourseCreatorRole.
```
Comment (johnbaldwin) [inline common/djangoapps/student/views/login.py:165]: Perhaps restate to "Return a user that matches the email only if the user has a course creator role"?

File: common/djangoapps/student/views/login.py [hunk 3]
```diff
@@ -153,6 +153,26 @@ def _log_failed_get_user_by_email(email):
...
+
+    This function should be only used in Studio.
```
Comment (johnbaldwin) [inline common/djangoapps/student/views/login.py:167]: I don't have a good idea/solution to this right now, but I'm just a bit concerned about a function that must not be called by the LMS that doesn't have built in protections against being called by the LMS.

I'm happy to discuss this, but I think the abstraction is happening in the wrong place. It is too deep down in "private" methods. We want to make this "Am I LMS or am I Studio" as high up as we can for visibility

Like here: https://github.com/appsembler/edx-platform/blob/appsembler/tahoe/master/common/djangoapps/student/views/login.py#L476

Main problem I see is that when a developer not familiar with the code goes to the logical place, `def login_user` to understand/debug/etc, the "I must be Studio" code is buried in a couple of layers of "private" functions.

I believe we really want to have the site type check/switch logic high up.

Also, we are special handling for Studio, for CSM but our site context is on the LMS, which can be confusing. It is sort of a "See this? Ok, DON'T do this. Instead, do this..."

I think changing the context to `is_in_cms` and then handling the special case in the `login_user` function would halve make the code abstraction clearer

File: common/djangoapps/student/views/login.py [hunk 4]
```diff
@@ -153,6 +153,26 @@ def _log_failed_get_user_by_email(email):
...
 
 
+def _is_in_lms():
+    """
+    Check if the function is executed in LMS.
+    """
+    return settings.ROOT_URLCONF == 'lms.urls'
+
+
+def _get_course_creator_by_email(email):
```
Comment (johnbaldwin) [inline common/djangoapps/student/views/login.py:163]: Does this really need to be a "private" function?

And I think it might benefit by restructuring to ask the question "is user a course creator"  so higher level functions can ask "does user exist and is user a course creator" which seems to me to be more readable

**Comment by johnbaldwin** [comment]
Awesome, now very visible, much better. Thanks @OmarIthawi !

**Comment by OmarIthawi** [comment]
@johnbaldwin could you please take another look?
